extractors/bookseries/karelians/extractors/tests/test_religion_extractor.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+from extractors.bookseries.karelians.extractors.religion_extractor import ReligionExtractor
+
+
+class TestReligionExtraction:
+
+    @pytest.yield_fixture(autouse=True)
+    def extractor(self, th):
+        return th.setup_extractor(ReligionExtractor(None, None))
+
+    def _verify_religion(self, expected_texts_and_years, extractor):
+        flag = 'religion'
+
+        for text, year in expected_texts_and_years:
+            results, metadata = extractor.extract({'text': text}, {}, {})
+
+            assert results[flag] == year
+
+
+    def should_correctly_extract_religion(self, extractor):
+        self._verify_religion([
+            ('Rouva X kuuluu ortodoksisen kirkon laulu-kuoroon ja Karjalaseuraan.', True)

```

Comment:
This example reminds me that maybe this extractor also requires some sex-detection logic so that the extracted data is assigned to the correct person? For example here the word Rouva refers clearly to the female.

I wonder if this would be a good place to experiment using conllu graphs to assign the data to the correct person? First find the keyword `ortodoksi` and then inspect the statement trying to find if there is a word indicating the identity of the person in question. In this case find out that word Rouva is related to word ortodoksi -> deduce that the female person of this text entry is in orthodoxian church.

This might well be out of scope of your current goals but might be an interesting and simple enough case to begin working with the more advanced text processing tools.

Reply:
Most of the hobbies, activities etc. keywords are troublesome in how to assign that data to the correct person entry and was one of the main reasons to start using the conllu data in extraction process. It might be good idea to develop some kind of utils for that first before going to production with extractors which do not have ability to discern the target person of the extracted data.